builder/build.go

```diff
@@ -170,7 +170,11 @@ func Build(pkgName, outpath string, config *compileopts.Config, action func(stri
 		for i, path := range config.ExtraFiles() {
 			abspath := filepath.Join(root, path)
 			outpath := filepath.Join(dir, "extra-"+strconv.Itoa(i)+"-"+filepath.Base(path)+".o")
-			err := runCCompiler(config.Target.Compiler, append(config.CFlags(), "-c", "-o", outpath, abspath)...)
+			targ:=config.Target.Triple
+			if targ!="" {
+				targ="--target="+config.Target.Triple
+			}
+			err := runCCompiler(config.Target.Compiler, append(config.CFlags(), "-c", targ,"-o", outpath, abspath)...)
```

Comment:
Instead of adding the `--target` flag here, the `Target` struct should be set up correctly. Assuming this is for the Raspberry Pi 3, adding it to the `cflags` key of the JSON file should be enough (if not, you can print `config.CFlags()` here to check whether `--target` is already included).

Reply:
Also note that this code does not affect the compilation of picolibc.

Reply:
The other change in this commit is the picolibc one.

This change with the --target is to make the smoketests pass. My config for rpi3 sets the target properly. Do you want me to investigate why it is not set during tests?

Reply:
Oh, I see what you mean. Took me a sec. I'll fix this to not use this mechanism and remove this from the PR.